src/main/actions/marktext.js

```diff
@@ -47,7 +47,7 @@ autoUpdater.on('update-downloaded', () => {
 })
 
 export const userSetting = (menuItem, browserWindow) => {
-  createWindow(userPreference.userDataPath)
+  userPreference.update().then(() => createWindow(userPreference.userDataPath))
```

Comment:
I think the best place to update user preference is the `Preference ` constructor.

Reply:
Yup, but it that safe (side effects; "write" is an non-blocking system call)?

One solution would be to load setting into cache and update local file. 

```
ctor {
  ...

  getAll()
  if (requiresUpdate) {
    this.update()
  }
}
```

Reply:
>side effects; "write" is an non-blocking system call

You mean `fs.write` ?

Reply:
Yes, in general system write.